content/en/docs/installing-on-linux/on-premises/install-kubesphere-on-vmware-vsphere.md

```diff
@@ -0,0 +1,401 @@
+
+![kubesphereOnVsphere-en-0-1-4-storage](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-4-storage.png)
+
+![kubesphereOnVsphere-en-0-1-5-compatibility](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-5-compatibility.png)
+
+![kubesphereOnVsphere-en-0-1-6-system](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-6-system.png)
+
+![kubesphereOnVsphere-en-0-1-7-hardware-1](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-7-hardware-1.png)
+
+![kubesphereOnVsphere-en-0-1-7-hardware-2](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-7-hardware-2.png)
+
+![kubesphereOnVsphere-en-0-1-7-hardware-3](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-7-hardware-3.png)
+
+![kubesphereOnVsphere-en-0-1-7-hardware-4](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-7-hardware-4.png)
+
+![kubesphereOnVsphere-en-0-1-8](/images/docs/vsphere/kubesphereOnVsphere-en-0-1-8.png)
+
+
+## 4. Keepalived+Haproxy
+###  1. Yum Install
+
+```bash
+#host lb-0(10.10.71.77) and host lb-1(10.10.71.66)
+yum install keepalived haproxy psmisc -y
+```
+
+### 2. Configure Haproxy
+
+On the servers with IP 10.10.71.77 and 10.10.71.66, configure haproxy (the configuration of the two lb machines is the same, pay attention to the back-end service address).
+```bash
+#Haproxy Configure /etc/haproxy/haproxy.cfg
+global
+    log         127.0.0.1 local2
+    chroot      /var/lib/haproxy
+    pidfile     /var/run/haproxy.pid
+    maxconn     4000
+    user        haproxy
+    group       haproxy
+    daemon
+    # turn on stats unix socket
+    stats socket /var/lib/haproxy/stats
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    log                     global
+    option                  httplog
+    option                  dontlognull
+    timeout connect         5000
+    timeout client          5000
+    timeout server          5000
+#---------------------------------------------------------------------
+# main frontend which proxys to the backends
+#---------------------------------------------------------------------
+frontend  kube-apiserver
+    bind *:6443
+    mode tcp
+    option tcplog
+    default_backend kube-apiserver
+#---------------------------------------------------------------------
+# round robin balancing between the various backends
+#---------------------------------------------------------------------
+backend kube-apiserver
+    mode tcp
+    option tcplog
+    balance     roundrobin
+    default-server inter 10s downinter 5s rise 2 fall 2 slowstart 60s maxconn 250 maxqueue 256 weight 100
+    server kube-apiserver-1 10.10.71.214:6443 check
+    server kube-apiserver-2 10.10.71.73:6443 check
+    server kube-apiserver-3 10.10.71.62:6443 check
+```
+```bash
+#Check for grammar before starting
```

Comment:
It would be better to divide the command and text into two lines, for example:

- Check for grammar before starting:

```
haproxy -f /etc/haproxy/haproxy.cfg -c
```

 Start Haproxy and set it to enable HAproxy:

```
systemctl restart haproxy && systemctl enable haproxy
```